src/util/readwritefile.h

```diff
@@ -11,18 +11,24 @@
 #include <string>
 #include <utility>
 
-/** Read full contents of a file and return them in a std::string.
- * Returns a pair <status, string>.
- * If an error occurred, status will be false, otherwise status will be true and the data will be returned in string.
+/**
+ * Read full contents of a file and return one of the following formats:
+ * 1. std::vector<unsigned char>
+ * 2. std::string
  *
- * @param maxsize Puts a maximum size limit on the file that is read. If the file is larger than this, truncated data
- *         (with len > maxsize) will be returned.
+ * @param[in] filename Filename. Returns false it doesn't exist.
+ * @param[out] output  Result
+ * @param[in] maxsize  Puts a maximum size limit on the file that is read. If the file
+ *                 is larger than this, truncated data (with len > maxsize) will be returned.
+ * @return true if successful, false if an error occured.
  */
-std::pair<bool,std::string> ReadBinaryFile(const fs::path &filename, size_t maxsize=std::numeric_limits<size_t>::max());
+template <typename T>
+bool ReadBinaryFile(const fs::path& filename, T& output, size_t maxsize=std::numeric_limits<size_t>::max());
```

Comment:
I had to change it because AFAIK you can't differentiate functions only by their return type. It's also more readable imo, no more `->first` ... `->second`.

`std::optional<T>` won't work, unless someone can explain how...

Reply:
> However, functions can not be overloaded if they differ only in the return type.

https://www.geeksforgeeks.org/function-overloading-and-return-type-in-cpp/

Reply:
I'll give that a try tomorrow.